mpp/shmem-def.h.in

```diff
@@ -74,7 +74,8 @@ extern "C" {
 
 #define SHMEM_MALLOC_ATOMICS_REMOTE   (1l<<0)
 #define SHMEM_MALLOC_SIGNAL_REMOTE    (1l<<1)
-#define SHMEM_MALLOC_MAX_HINTS        ((1l<<2) - 1)
+#define SHMEMX_MALLOC_NO_BARRIER      (1l<<2)
```

Comment:
As discussed on a call, this would split up the hints, which is not ideal, but a comment here would help.  Something like, `/*  MAX_HINTS value includes the SHMEMX constant(s) in shmemx.h4`

Reply:
How did you decide that this hint should be moved to `mpp/shmemx.h4` instead of `mpp/shmemx-def.h`? There is a note at the top of `mpp/shmemx.h4` that reads "This is a generated file, do not edit directly", which makes me think that the hint should be moved to `mpp/shmemx-def.h`.

Reply:
Is there another file that I am supposed to update which feeds into the generation of mpp/shmemx.h4?

Reply:
Thanks for catching that comment @kholland-intel!  It's troubling especially when I discovered that our `.gitignore` file includes `mpp/shmemx.h4`!  That's not good... :)

I looked into it and think I have it figured out.  If you follow the git blame and logs, you'll see that `mpp/shmemx.h4.in` was the initial file used to generate `mpp/shmemx.h4` and this comment must have been carried along through the transition in commit https://github.com/Sandia-OpenSHMEM/SOS/commit/87f3de7ee646720e49f9572e0ff4b3b3991ee0cc, which otherwise simplified the header generation.

So I think we should remove this comment, also remove the comment from `pshmem.h4`, and remove both of these files from `.gitignore`.  And I still think we should put new constants in `shmemx.h4`, because that file becomes `shmemx.h` which is installed for users and we want them to be able to see the constant.  Make sense?

Reply:
Ah ok, this makes sense now. Thanks for tracking this down! 👍